syft/scope/resolvers/directory_resolver.go

```diff
@@ -82,9 +82,20 @@ func (s DirectoryResolver) MultipleFileContentsByRef(f ...file.Reference) (map[f
 		contents, err := fileContents(fileRef.Path)
 
 		if err != nil {
-			return refContents, fmt.Errorf("could not read contents of file: %s", fileRef.Path)
+			return nil, fmt.Errorf("could not read contents of file: %s", fileRef.Path)
```

Comment:
not sure I follow this change, mind expanding why is this required?

Reply:
not required, just an improvement along the way. When returning errors it's good practice to not return partial information, but instead blank or nil values.

Reply:
Aha! good to know, I tend to get confused about being strict on what is returned and using `nil`